app/controllers/users_controller.rb

```diff
@@ -2,26 +2,54 @@
 
 class UsersController < ApplicationController
   include TokenHelper
-  # rubocop: disable Style/GuardClause
-  def signup
-    user = User.new(user_params)
-    if user.save
-      token = generate_token({ exp: (Time.now + 1800).to_i, user_id: user.id })
-      response.status = 201
-      render json: {
-        data: {
-          token: token
-        }
+  rescue_from ActionController::ParameterMissing, with: :missing_parameter
+
+  def signin
+    user = User.find_by(email: user_params([:email, :password])[:email])
+    unless user&.authenticate(params[:user][:password])
```

Comment:
Could refactor to use rescue from <certain error> with a private method - one for user not found (user `find_by!` i.e. with the bang), and one for the error thrown by bcrypt's `authenticate`, then could move the large 401 block into those methods and return rubocop config back to normal.